docs/blog/translation-python37-new-features/index.md

```diff
@@ -16,6 +16,139 @@
 
 ## 향상된 타이핑
 
+타입 힌팅과 어노테이션은 Python 3 릴리즈 전체에서 꾸준히 개발되어 왔습니다.
+파이썬의 <a href="https://www.youtube.com/watch?v=2xWhaALHTvU" target="_blank">타이핑 시스템</a>은 이제 꽤 안정적입니다.
+그럼에도 불구하고 Python 3.7은 성능 향상, 핵심 지원 및 전방 참조(forward references)와 같은 몇 가지 개선 사항을 제공합니다.
+
+파이썬은 런타임 시에 (명시적으로 <a href="https://pypi.org/project/enforce/" target="_blank">enforce</a>와 같은 패키지를 사용하지 않는 한) 타입 검사를 하지 않습니다.
+따라서 코드에 타입 힌트를 추가해도 성능에 영향을 미치지 않습니다.
+
+불행히도, 대부분의 타입 힌트가 `typing` 모듈을 필요로 하기 때문에 완전하지 않습니다.
+`typing` 모듈은 표준 라이브러리에서 <a href="https://www.python.org/dev/peps/pep-0560/#performance" target="_blank">가장 느린 모듈</a> 중 하나입니다.
+<a href="https://www.python.org/dev/peps/pep-0560/" target="_blank">PEP 560</a>은 `typing` 모듈의 속도를 대폭 향상시키는 Python 3.7에서 타이핑에 대한 핵심 지원을 추가합니다.
+이것에 대한 세부 사항은 일반적으로 알 필요가 없습니다. 느긋하게 누워서 향상된 성능을 즐기세요.(TODO: 더 자연스러운 표현은 없을까)
+
+파이썬의 타입 시스템은 합리적으로 표현력이 있지만, 약간의 고통을 주는 한 가지 문제는 전방 참조(forward references)입니다.
+타입 힌트(또는 일반적으로 어노테이션)는 모듈을 가져 오는 동안 계산됩니다.
+따라서 모든 이름은 사용되기 전에 정의되어 있어야합니다.
+따라서 다음은 불가능합니다:
+
+```python
+class Tree:
+    def __init__(self, left: Tree, right: Tree) -> None:
+        self.left = left
+        self.right = right
+```
+
+클래스 `Tree`가 `.__init__()` 메소드에서 아직 (완전히) 정의되지 않았으므로 코드를 실행하면 `NameError`가 발생합니다.
+
+```python
+Traceback (most recent call last):
+  File "tree.py", line 1, in <module>
+    class Tree:
+  File "tree.py", line 2, in Tree
+    def __init__(self, left: Tree, right: Tree) -> None:
+NameError: name 'Tree' is not defined
+```
+
+이를 해결하기 위해 문자열 리터럴 `"Tree"`를 써야합니다.
+
+```python
+class Tree:
+    def __init__(self, left: "Tree", right: "Tree") -> None:
+        self.left = left
+        self.right = right
+```
+
+원래 토론에 대해서는 <a href="https://www.python.org/dev/peps/pep-0484/#forward-references" target="_blank">PEP 484</a>를 참조하세요.
+
+미래의 <a href="http://www.curiousefficiency.org/posts/2014/08/python-4000.html" target="_blank">Python 4.0</a>에서는 이와 같이 전방 참조(forward references)가 허용될 것입니다.
+이는 명시적으로 요구될 때까지 어노테이션을 계산하지 않음으로써 처리됩니다.
+<a href="https://www.python.org/dev/peps/pep-0563/" target="_blank">PEP 563</a>은 이 제안의 세부 사항을 설명합니다.
+Python 3.7에서 전방 참조는 이미 <a href="https://docs.python.org/3/library/__future__.html" target="_blank">__future__</a>를 `import` 해서 사용 가능합니다.
+이제 다음과 같이 작성할 수 있습니다:
+
+```python
+from __future__ import annotations
+
+class Tree:
+    def __init__(self, left: Tree, right: Tree) -> None:
+        self.left = left
+        self.right = right
+```
+
+다소 불명확한 `"Tree"` 구문을 피하는 것 외에도 타입 힌트 계산이 지연되지 않으므로 지연된 어노테이션 계산도 코드 속도를 높입니다.(TODO: 말이 너무 꼬인 느낌)
```

Comment:
원문: `Note that in addition to avoiding the somewhat clumsy "Tree" syntax, the postponed evaluation of annotations will also speed up your code, since type hints are not executed. `